Истинное значение(y_true) : 110, 135, 90, 89, 101, 145, 177, 90, 88, 105


Предсказание модели(y_pred) : 105, 110, 87, 94, 98, 122, 150, 89, 84, 102

In [ ]:
y_true = [110, 135, 90, 89, 101, 145, 177, 90, 88, 105]

In [ ]:
y_pred = [105, 110, 87, 94, 98, 122, 150, 89, 84, 102]

1. Рассчитайте показатель «среднеквадратическая ошибка» (Mean Squared Error – MSE) 

In [ ]:
from sklearn.metrics import mean_squared_error
MSE = mean_squared_error(y_true, y_pred)
MSE

197.7

2. Рассчитайте показатель «корень среднеквадратической ошибки» (Root Mean Squared Error – RMSE) 

In [ ]:
from sklearn.metrics import mean_squared_error
RMSE = mean_squared_error(y_true, y_pred, squared = False)
RMSE

14.060583202698243

3. Рассчитайте показатель «средний модуль ошибки» (Mean Absolute Error – MAE)) 

In [ ]:
from sklearn.metrics import mean_absolute_error
MAE = mean_absolute_error(y_true, y_pred)
MAE

9.9

4. Рассчитайте коэффициент детерминации (R2) 

In [ ]:
from sklearn.metrics import r2_score
R2 = r2_score(y_true, y_pred)
R2

0.7553217821782179

5. Сформулируйте и аргументируйте вывод о точности построенной банком регрессионной модели.

In [ ]:
print("MAE: {0:7.2f}, RMSE: {1:7.2f}, r2_score{3:7.2f} для Модели линейной регрессии".format(MSE, RMSE, MAE, R2))

MAE:  197.70, RMSE:   14.06, r2_score   0.76 для Модели линейной регрессии


Вывод: модель среднего качества, она оценивает на 76% вариативность нашей выборки и у нее достаточно большие ошибки.